pyfarm/master/user_interface/jobgroups.py

```diff
@@ -0,0 +1,242 @@
+    tasks_failed_query = db.session.query(
+        func.count(Task.id).label('t_failed'), Job.job_group_id).\
+            join(Job, Task.job_id == Job.id).\
+            filter(Task.state == WorkState.FAILED).\
+                group_by(Job.job_group_id).subquery()
+    jobgroups_query = db.session.query(JobGroup,
+                                       User.username,
+                                       JobType.name.label("main_jobtype_name"),
+                                       func.coalesce(
+                                           agent_count_query.c.agent_count,
+                                           0).label("agent_count"),
+                                       submit_time_query.c.time_submitted.\
+                                           label("time_submitted"),
+                                       func.coalesce(
+                                           tasks_queued_query.c.t_queued,
+                                           0).label("t_queued"),
+                                       func.coalesce(
+                                           tasks_running_query.c.t_running,
+                                           0).label("t_running"),
+                                       func.coalesce(
+                                           tasks_done_query.c.t_done,
+                                           0).label("t_done"),
+                                       func.coalesce(
+                                           tasks_failed_query.c.t_failed,
+                                           0).label("t_failed")
+                                       ).\
+        join(JobType, JobGroup.main_jobtype_id == JobType.id).\
+        outerjoin(jobs_queued_query,
+                  JobGroup.id == jobs_queued_query.c.job_group_id).\
+        outerjoin(jobs_paused_query,
+                  JobGroup.id == jobs_paused_query.c.job_group_id).\
+        outerjoin(jobs_running_query,
+                  JobGroup.id == jobs_running_query.c.job_group_id).\
+        outerjoin(jobs_done_query,
+                  JobGroup.id == jobs_done_query.c.job_group_id).\
+        outerjoin(jobs_failed_query,
+                  JobGroup.id == jobs_failed_query.c.job_group_id).\
+        outerjoin(tasks_queued_query,
+                  JobGroup.id == tasks_queued_query.c.job_group_id).\
+        outerjoin(tasks_running_query,
+                  JobGroup.id == tasks_running_query.c.job_group_id).\
+        outerjoin(tasks_done_query,
+                  JobGroup.id == tasks_done_query.c.job_group_id).\
+        outerjoin(tasks_failed_query,
+                  JobGroup.id == tasks_failed_query.c.job_group_id).\
+        outerjoin(User, JobGroup.user_id == User.id).\
+        outerjoin(agent_count_query,
+                  JobGroup.id == agent_count_query.c.job_group_id).\
+        outerjoin(submit_time_query,
+                  JobGroup.id == submit_time_query.c.job_group_id)
+
+    filters = {}
```

Comment:
nit: could this be updated to be build something like this:

```python
filters = {
    "st_queued": ...,
    "st_paused": ...
}
```

I think it's easier to read/update.